misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BTable.java

```diff
@@ -73,8 +77,8 @@ public String computeValue() {
             if (!(jvmValue instanceof ObjectReference)) {
                 return Either.forRight(new ArrayList<>());
             }
-            Value[] tableKeys = getTableKeys(start, count);
-            return Either.forRight(getTableEntriesFor(tableKeys));
+            populateTableValues(start, count);
+            return Either.forRight(getChildVariables(start, count));
```

Comment:
This logic does not consider using the already loaded table values(entries). Please modify the logic to immediately return if the requested value range is already loaded and available in the `tableValues`

Reply:
The mentioned logic is already added inside `populateTableValues()` method.
https://github.com/ballerina-platform/ballerina-lang/blob/7f54dc568b0f75708932c68aaa2695f7bea2afc4/misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BTable.java#L145

Reply:
But the while loop does not end even after all the required variables are loaded, because of https://github.com/ballerina-platform/ballerina-lang/blob/7f54dc568b0f75708932c68aaa2695f7bea2afc4/misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BTable.java#L142

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/30994/commits/91daed1acb68ec02ead65479dfa4994f08cdf4c7